docs/articles/apex/writing.md

```diff
@@ -26,51 +37,58 @@ You can preview, view, or go to definitions of:
   - Variables (local and class variables)
 - Standard objects
   - Fields (standard and custom fields)
-  - Object definitions
+  - Object definition
 - Custom objects
   - Fields
   - Object definitions
 
-(See the [Enable Code Smartness for SObjects](#enable-code-smartness-for-sobjects) section of this topic for information on working with standard and custom objects.)
+When you refresh SObject definitions, the extension uses the default org to generate Apex classes. These classes represent the standard and custom objects that the current user has access to. **Do not** edit these representative Apex classes, which are stored in a hidden directory on your local workstation.
+
+Whenever you refresh SObject definitions, the representative Apex classes are deleted and regenerated. You can modify the SObjects either by updating the objects’ `.object-meta.xml` and `.field-meta.xml` files; Or by making changes declaratively in the default org. After modifying the SObjects, make sure to sync your local project and the default org.
 
-To preview a definition, hold down Ctrl (Windows or Linux) or Cmd (macOS) and hover over the item whose definition you want to see.
+When you launch the Salesforce CLI Integration extension (which is part of the Salesforce Extension Pack) for the first time, `SFDX: Refresh SObject Definitions` command is executed automatically if your project doesn’t contain any Apex classes. After you add or edit standard or custom objects or their fields, make sure to run this command.
```

Comment:
This is currently not true. For sobject refresh to run automatically the user needs to enable the `salesforcedx-vscode-apex.enable-sobject-refresh-on-startup` setting.